### ISSUE TITLE
more than one bond between consecutive cystines 

### ISSUE THREAD

**Issue by rwxayheee** [forlilab/Meeko#213]
The protein (PDB ID 1UW6) used in Vina hydrated docking tutorial is a special case where there are consecutive CYX. They're not only connected by the backbone but also the disulfide bond. 

Input file: 
[1uw6_receptorH.pdb.txt](https://github.com/user-attachments/files/17518093/1uw6_receptorH.pdb.txt)

Command

```
mk_prepare_receptor.py -i 1uw6_receptorH.pdb -o 1uw6_receptor -j -p
```

Traceback

```
Traceback (most recent call last):
  File "/Users/amyhe/micromamba/envs/has_espaloma/bin/mk_prepare_receptor.py", line 7, in <module>
    exec(compile(f.read(), __file__, 'exec'))
  File "/Users/amyhe/Desktop/0_forks/Meeko/scripts/mk_prepare_receptor.py", line 512, in <module>
    chorizo = LinkedRDKitChorizo.from_prody(
  File "/Users/amyhe/Desktop/0_forks/Meeko/meeko/linked_rdkit_chorizo.py", line 1021, in from_prody
    chorizo = cls(
  File "/Users/amyhe/Desktop/0_forks/Meeko/meeko/linked_rdkit_chorizo.py", line 749, in __init__
    raise ValueError(msg)
ValueError: got more than one bond between some residue pairs: ('A:187', 'A:188') ('B:187', 'B:188')
```

**Comment by rwxayheee** [comment]
It seems like we can allow more than one bonds between residue pairs... Skipping the check block will allow the structure to pass

**Comment by diogomart** [comment]
Great catch. I was lazy and stored the bonds in a dictionary where the key is a pair of residue IDs, and the value is a pair of atom indices. Will need to make the value a list of pairs of atom indices. Then we can remove the check.

**Comment by diogomart** [comment]
You are right, it's OK to bypass the check after a few fixes in the code in 1ea7eb434120ead7583c4a80fc7dc06db06ffdc1

The only limitation of 1ea7eb434120ead7583c4a80fc7dc06db06ffdc1 is that the second and subsequent bonds between a pair of residues, which in this case is the S-S bond between adjacent Cys, does not get the coordinate assignment for the padded atoms. Thus, the -S-CH3 that pads each cys has all coordinates set to zeros. The same happens for all blunt ends. This is OK.

I'll open a new issue in case we find time to fix it in the future.

**Comment by rwxayheee** [comment]
Sounds good, thanks!!

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________
